contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -1588,6 +1590,234 @@ create_xp_instance_regread_in_master_dbo_internal(PG_FUNCTION_ARGS)
+							d_schema = defel;
+							schemaName = defGetString(d_schema);
+						}
+						if (strcmp(defel->defname, "cascade") == 0)
+						{
+							ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+									errmsg("'cascade' is not yet supported in Babelfish.")));
+						}
+					}
+
+					if(schemaName != NULL && (strcmp(schemaName, "sys") && strcmp(schemaName, "public")))
+					{
+						ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+									errmsg("'%s' is not a valid schema, only sys and public schema is allowed in Babelfish.", schemaName)));
+					}
+					
+					for(int i = 0; i < allowed_extns_size; i++)
+					{
+						if((strcmp(crstmt->extname, allowed_extns[i])))
+						{
+							if(i == allowed_extns_size - 1)
+							{
+								ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+									errmsg("'%s' is not a valid name.", crstmt->extname)));
+							}
+							else
+							{
+								continue;
+							}
+						}
+						else
+						{
+							break;
+						}
+					}
+					/* do this step */
+					ProcessUtility(wrapper,
+								extensionStmt,
+								false,
+								PROCESS_UTILITY_SUBCOMMAND,
+								NULL,
+								NULL,
+								None_Receiver,
+								NULL);
+
+					/* make sure later steps can see the object created here */
+					 CommandCounterIncrement();
+					break;
+				}
+				case T_DropStmt:
+				{
+					DropStmt *drstmt = (DropStmt *) parsetree;
+
+					if(drstmt->behavior == DROP_CASCADE)
+					{
+						ereport(ERROR, (errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+									errmsg("'cascade' is not yet supported in Babelfish.")));
+					}
+
+					if (drstmt->removeType == OBJECT_EXTENSION)
+					{
+						/* do this step */
+						ProcessUtility(wrapper,
+									extensionStmt,
+									false,
+									PROCESS_UTILITY_SUBCOMMAND,
+									NULL,
+									NULL,
+									None_Receiver,
+									NULL);
+
+						/* make sure later steps can see the object created here */
+						CommandCounterIncrement();
+					}
+					break;
+				}
+				case T_AlterExtensionStmt:
+				{
+					/* do this step */
+					ProcessUtility(wrapper,
+								extensionStmt,
+								false,
+								PROCESS_UTILITY_SUBCOMMAND,
+								NULL,
+								NULL,
+								None_Receiver,
+								NULL);
+
+					/* make sure later steps can see the object created here */
+					 CommandCounterIncrement();
+					break;
+				} 
+				case T_AlterObjectSchemaStmt:
+				{
+					ereport(ERROR,
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					 errmsg("Alter extension schema is not currently supported in Babelfish")));
+					break;
+				}
+				case  T_AlterExtensionContentsStmt:
+				{
+					ereport(ERROR,
+					(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+					 errmsg("Alter extension to Add/Drop object in extension is not currently supported in Babelfish")));
+					break;
+				}
+				default:
+				elog(ERROR, "unrecognized node type: %d",
+					 (int) nodeTag(parsetree));
+				break;
+			}
+		}
+		SetCurrentRoleId(current_user_id, false);
```

Comment:
We should be including it in PG_CATCH() and after end of the try. Just how we are resetting the config.